### PR TITLE
Adding Enzyme

### PR DESCRIPTION
**Pull request by IvanYashchuk**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by IvanYashchuk** [comment]
@conda-forge-admin, please ping conda-forge/help-c-cpp

File: recipes/libenzyme/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# Needed for the linter even though it's specified in conda_build_config.yaml
+{% set llvm_major_version = llvm_major_version or "11" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/wsmoses/Enzyme/archive/v{{ version }}.tar.gz
+  sha256: ce15ce1b839ba68e0941ce39730ab4df938eb1a6383feaffbccd3276fac22d14
```
[the remark this answers is not in the document]
Comment (IvanYashchuk) [reply recipes/libenzyme/meta.yaml:14]: The version is not arbitrary, it's fixed by convention at the beginning of the file `{% set version = "0.0.23" %}`. When the version changes, sha256 would need to be updated as well.

[the remark this answers is not in the document]
Comment (wsmoses) [reply recipes/libenzyme/meta.yaml:14]: Ah, got it

**Review by carterbox — CHANGES_REQUESTED** [review]

**Comment by carterbox** [comment]
@conda-forge-admin, please restart ci

File: recipes/libenzyme/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    - lit
+  run:
+    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx and x86_64]
+    - {{ "libllvm" + llvm_major_version }}
+  run_constrained:
+    - llvmdev {{ llvm_major_version + ".*" }}
+
+test:
+  commands:
+    - test -f $PREFIX/lib/LLVMEnzyme-{{ llvm_major_version }}${SHLIB_EXT}  # [unix]
```
Comment (wsmoses) [inline recipes/libenzyme/meta.yaml:40]: Can you add a test here for ClangEnzyme and LLDEnzyme in addition to LLVMEnzyme?

File: recipes/libenzyme/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    The Enzyme project is a tool that takes arbitrary existing code as
+    LLVM IR and computes the derivative (and gradient) of that function.
+    This allows developers to use Enzyme to automatically create gradients
+    of their source code without much additional work.
+  doc_url: https://enzyme.mit.edu/
+  dev_url: https://github.com/wsmoses/Enzyme
+
+extra:
+  recipe-maintainers:
+    - IvanYashchuk
```
Comment (wsmoses) [inline recipes/libenzyme/meta.yaml:57]: Feel free to add myself here

@vchuravy @tgymnich may also want to be added

File: recipes/libenzyme/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  skip: True  # [win]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+  host:
+    - {{ "libllvm" + llvm_major_version }}
+    - llvmdev {{ llvm_major_version }}
```
Comment (wsmoses) [inline recipes/libenzyme/meta.yaml:29]: It's likely worth adding clangdev here so the nicer helper ClangEnzyme-XX.so gets built too.

**Pull request closed without merging**